### PR TITLE
[clang-tidy] remove needless std::move

### PR DESCRIPTION
**Pull request by neheb**
As a result of the last commit.

Found with performance-move-const-arg

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: src/storage/plugins/UdisksStorage.cxx [hunk 1]
```diff
@@ -168,7 +168,7 @@ UdisksStorage::OnListReply(ODBus::Message reply) noexcept
 			if (!o.IsId(id))
 				return;
 
-			dbus_path = std::move(o.path);
+			dbus_path = o.path;
```
Comment (MaxKellermann) [inline src/storage/plugins/UdisksStorage.cxx]: This change is bad. You have let clang-tidy do its job without understanding the real problem.

File: src/storage/CompositeStorage.cxx [hunk 2]
```diff
@@ -118,7 +118,7 @@ CompositeStorage::Directory::Make(const char *uri)
 	Directory *directory = this;
 	while (*uri != 0) {
 		const std::string name = NextSegment(uri);
-		auto i = directory->children.emplace(std::move(name),
+		auto i = directory->children.emplace(name,
```
Comment (MaxKellermann) [inline src/storage/CompositeStorage.cxx]: Same here. The problem is the `const`, not the `std::move()`.

File: src/config/File.cxx [hunk 3]
```diff
@@ -72,7 +72,7 @@ config_read_name_value(ConfigBlock &block, char *input, unsigned line)
 		throw FormatRuntimeError("\"%s\" is duplicate, first defined on line %i",
 					 name, bp->line);
 
-	block.AddBlockParam(name, std::move(value), line);
+	block.AddBlockParam(name, value, line);
```
Comment (MaxKellermann) [inline src/config/File.cxx]: Why remote this `std::move()`?

Comment (neheb) [reply src/config/File.cxx]: warning: std::move of the variable 'value' of the trivially-copyable type 'char *' has no effect; remove std::move() [hicpp-move-const-arg]

File: src/storage/CompositeStorage.cxx (hunk 2 above)
Comment (neheb) [reply src/storage/CompositeStorage.cxx]: Warning: std::move of the const variable 'name' has no effect; remove std::move() or make the variable non-const [hicpp-move-const-arg]

I guess I  can remove the const.

File: src/storage/plugins/UdisksStorage.cxx (hunk 1 above)
Comment (neheb) [reply src/storage/plugins/UdisksStorage.cxx]: warning: std::move of the const expression has no effect; remove std::move() [hicpp-move-const-arg]

File: src/storage/CompositeStorage.cxx (hunk 2 above)
Comment (neheb) [reply src/storage/CompositeStorage.cxx]: Fixed.

**Comment by neheb** [comment]
clang-tidy claims to remove std::move under three conditions:

if std::move() is called with a constant argument,
if std::move() is called with an argument of a trivially-copyable type,
if the result of std::move() is passed as a const reference argument.

All are shown when running the tool.

File: src/storage/plugins/UdisksStorage.cxx (hunk 1 above)
Comment (MaxKellermann) [reply src/storage/plugins/UdisksStorage.cxx]: The warning is correct, but not the solution. That's what I mean: you use clang-tidy blindly, without understanding why the problem is there, and let clang-tidy choose a solution which isn't good here.

**Comment by MaxKellermann** [comment]
ac50bb5d2bbe1dad81eac6a6fd84d1f0d25eccab

I removed several wrong changes.

**Pull request closed without merging**